azure-kusto-ingest/azure/kusto/ingest/resource_manager.py

```diff
@@ -0,0 +1,99 @@
+"""
+    This module is serve as a cache to all resources needed by the kusto ingest client
+"""
+
+from azure.kusto.data import KustoClient
+from .connection_string import _ConnectionString
+
+from datetime import datetime
+from datetime import timedelta
+import random
+
+class IngestClientResources:
+    def __init__(self,
+                 secured_ready_for_aggregation_queues = None,
+                 failed_ingestions_queues = None,
+                 successful_ingestions_queues = None,
+                 containers = None,
+                 status_table = None
+                 ):
+        self.secured_ready_for_aggregation_queues = secured_ready_for_aggregation_queues
+        self.failed_ingestions_queues = failed_ingestions_queues
+        self.successful_ingestions_queues = successful_ingestions_queues
+        self.containers = containers
+        self.status_table = status_table
+    
+    def is_applicable(self):
+        if self.secured_ready_for_aggregation_queues \
+           and self.failed_ingestions_queues \
+           and self.successful_ingestions_queues \
+           and self.containers \
+           and self.status_table:
+            return True
+        return False
+
+class _ResourceManager:
+    def __init__(self, 
+                 kusto_client):
+        self._kusto_client = kusto_client
+        self._cache_lifetime = timedelta(hours = 5)
+
+        self._ingest_client_resources = None
+        self._ingest_client_resources_last_updtae = None
+
+        self._authorization_context = None
+        self._authorization_context_last_updtae = None
+
+    def _refresh_ingest_client_resources(self):
+        if not self._ingest_client_resources \
+           or self._ingest_client_resources_last_updtae + self._cache_lifetime <= datetime.now() \
```

Comment:
Store the calculation in a variable and then check the value.

You also have a variable name typo: `_ingest_client_resources_last_updtae`. (Are there tests for this?)

Reply:
I rather leave the calculation within the check in order to make the code purpose more clear.
Tests are added in a different pull request, that will be out soon.
Also changed all the typo, thanks.